tests/test_pattern_rewriter.py

```diff
@@ -540,27 +538,31 @@ def match_and_rewrite(self, op: ModuleOp, rewriter: PatternRewriter):
 def test_block_argument_type_change():
     """Test the modification of a block argument type."""
 
-    prog = """"builtin.module"() ({
-  %0 = "arith.constant"() {"value" = true} : () -> i1
-  "scf.if"(%0) ({
-  ^0(%1 : i32):
-    %2 = "arith.addi"(%1, %1) : (i32, i32) -> i32
-  }, {}) : (i1) -> ()
-}) : () -> ()"""
+    prog = """\
+"builtin.module"() ({
+  %0 = "test.op"() ({
+  ^0(%1 : !test.type<"int">):
+    %2 = "test.op"() : () -> !test.type<"int">
+  }) : () -> !test.type<"int">
+}) : () -> ()
+"""
 
-    expected = """"builtin.module"() ({
-  %0 = "arith.constant"() {"value" = true} : () -> i1
-  "scf.if"(%0) ({
+    expected = """\
+"builtin.module"() ({
+  %0 = "test.op"() ({
   ^0(%1 : i64):
-    %2 = "arith.addi"(%1, %1) : (i64, i64) -> i32
-  }, {
-  }) : (i1) -> ()
-}) : () -> ()"""
+    %2 = "test.op"() : () -> !test.type<"int">
+  }) : () -> !test.type<"int">
+}) : () -> ()
+"""
 
     class Rewrite(RewritePattern):
         @op_type_rewrite_pattern
-        def match_and_rewrite(self, op: If, rewriter: PatternRewriter):
-            rewriter.modify_block_argument_type(op.true_region.blocks[0].args[0], i64)
+        def match_and_rewrite(self, matched_op: TestOp, rewriter: PatternRewriter):
+            if matched_op.regs and matched_op.regs[0].blocks:
+                rewriter.modify_block_argument_type(
+                    matched_op.regs[0].blocks[0].args[0], i64
+                )
```

Comment:
Not sure how others feel, but would it make sense to had a RegOp in the testing dialect, that has arbitrarily many regions? This way, you could get rid of this if, which feels not quite as nice to me.

Reply:
Nice idea.

Reply:
Personally, I think for rewriting with the testing dialect, it might make sense to add these structures. I would make sure to restrict ourselves to very few, but I don't think a couple more operations (with clear purpose) makes the testing dialect too big. What do you think, @math-fehr?

Reply:
And @superlopuh I guess

Reply:
Seems consistent, would probably make life a little easier. It's the sort of thing I'd do as a subtask of a task, not as a separate goal TBH.

Reply:
@superlopuh I know what you mean, but I can add it separately in another PR and apply it here, since there is no immediate need for it.
I'd rather not stall this PR for that, but I don't really mind, so up to you guys.

Reply:
Sure, we can merge this and open an issue or something.